src/cairoUtilFuncGen/inputArgCheck/inputCheck.ts

```diff
@@ -0,0 +1,267 @@
+      return this.createIntInputCheck(type);
+    } else if (type instanceof BoolType) {
+      return this.createBoolInputCheck();
+    } else if (type instanceof UserDefinedType && type.definition instanceof StructDefinition) {
+      return this.createStructInputCheck(key, type);
+    } else if (type instanceof UserDefinedType && type.definition instanceof EnumDefinition) {
+      return this.createEnumInputCheck(key, type);
+    } else if (type instanceof ArrayType) {
+      return type.size === undefined
+        ? this.createDynArrayInputCheck(key, type)
+        : this.createStaticArrayInputCheck(key, type);
+    } else if (type instanceof AddressType) {
+      return this.createAddressInputCheck();
+    } else {
+      throw new NotSupportedYetError(` Input check for ${printTypeNode(type)} not defined yet.`);
+    }
+  }
+
+  private createIntInputCheck(type: IntType): string {
+    const funcName = `warp_external_input_check_int${type.nBits}`;
+    this.requireImport(
+      'warplib.maths.external_input_check_ints',
+      `warp_external_input_check_int${type.nBits}`,
+    );
+    return funcName;
+  }
+
+  private createAddressInputCheck(): string {
+    const funcName = 'warp_external_input_check_address';
+    this.requireImport(
+      'warplib.maths.external_input_check_address',
+      `warp_external_input_check_address`,
+    );
+    return funcName;
+  }
+
+  private createStructInputCheck(key: string, type: UserDefinedType): string {
+    const funcName = `extern_input_check${this.generatedFunctions.size}`;
+    const implicits = '{range_check_ptr : felt}';
+
+    // Set an empty entry so recursive function generation doesn't clash
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
+    const structDef = type.definition;
+    assert(structDef instanceof StructDefinition);
+    const cairoType = CairoType.fromSol(type, this.ast, TypeConversionContext.CallDataRef);
+
+    this.generatedFunctions.set(key, {
+      name: funcName,
+      code: [
+        `func ${funcName}${implicits}(arg : ${cairoType.toString()}) -> ():`,
+        `alloc_locals`,
+        ...structDef.vMembers.map((decl) => {
+          const memberType = getNodeType(decl, this.ast.compilerVersion);
+          this.checkForImport(memberType);
+          if (this.checkableType(memberType)) {
+            const memberCheck = this.getOrCreate(memberType);
+            return [`${memberCheck}(arg.${decl.name})`];
+          } else {
+            return '';
+          }
+        }),
+        `return ()`,
+        `end`,
+      ].join('\n'),
+    });
+    return funcName;
+  }
+
+  private createStaticArrayInputCheck(key: string, type: ArrayType): string {
+    const funcName = `extern_input_check${this.generatedFunctions.size}`;
+    const implicits = '{range_check_ptr : felt}';
+
+    // Set an empty entry so recursive function generation doesn't clash
+    this.generatedFunctions.set(key, { name: funcName, code: '' });
```

Comment:
All the instances of doing this can be moved into getOrCreate so there's only one and no chance of missing one